src/brokerimpl.rs

```diff
@@ -857,21 +857,31 @@ impl BrokerState {
                 None
             }
         };
+
+        if let Some(mount_point) = effective_mount_point.as_ref() {
+            if let Some(mount) = self.mounts.get(mount_point) {
+                sender.close();
```

Comment:
Why do you need to close the channel before returning from the function? Isn't the drop behavior sufficient?

Reply:
right now the sender is (wrongly) held in the peer loop too, so just dropping is not sufficient. I have another set of patched which fix this stuff, so please let this slide for now, and it'll be fixed in the next PR 🙏